pkg/user/tx_client.go

```diff
@@ -468,40 +416,67 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 			return nil, err
 		}
 
-		if resp != nil {
-			switch resp.Status {
-			case core.TxStatusPending:
-				// Continue polling if the transaction is still pending
-				select {
-				case <-ctx.Done():
-					return nil, ctx.Err()
-				case <-pollTicker.C:
-					continue
-				}
-			case core.TxStatusCommitted:
-				txResponse := &TxResponse{
-					Height: resp.Height,
-					TxHash: txHash,
-					Code:   resp.ExecutionCode,
-				}
-				if resp.ExecutionCode != abci.CodeTypeOK {
-					executionErr := &ExecutionError{
-						TxHash:   txHash,
-						Code:     resp.ExecutionCode,
-						ErrorLog: resp.Error,
-					}
-					return nil, executionErr
+		switch resp.Status {
+		case core.TxStatusPending:
+			// Continue polling if the transaction is still pending
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			case <-pollTicker.C:
+				continue
+			}
+		case core.TxStatusCommitted:
+			txResponse := &TxResponse{
+				Height: resp.Height,
+				TxHash: txHash,
+				Code:   resp.ExecutionCode,
+			}
+			if resp.ExecutionCode != abci.CodeTypeOK {
+				executionErr := &ExecutionError{
+					TxHash:   txHash,
+					Code:     resp.ExecutionCode,
+					ErrorLog: resp.Error,
 				}
-				return txResponse, nil
-			case core.TxStatusEvicted:
-				return nil, fmt.Errorf("tx was evicted from the mempool")
-			default:
-				return nil, fmt.Errorf("unknown tx: %s", txHash)
+				client.deleteFromLocalMempool(txHash)
+				return nil, executionErr
 			}
+			client.deleteFromLocalMempool(txHash)
+			return txResponse, nil
+		case core.TxStatusEvicted:
+			return nil, client.handleEvictions(txHash)
+		default:
+			client.deleteFromLocalMempool(txHash)
+			return nil, fmt.Errorf("transaction with hash %s not found; it was likely rejected", txHash)
 		}
 	}
 }
 
+// handleEvictions handles the scenario where a transaction is evicted from the mempool.
+// It removes the evicted transaction from the local pool without incrementing
+// the signer's sequence.
+func (client *TxClient) handleEvictions(txHash string) error {
+	client.mtx.Lock()
+	defer client.mtx.Unlock()
+	// Get transaction from the local pool
+	txInfo, exists := client.localMempool[txHash]
+	if !exists {
+		return fmt.Errorf("tx not found in tx client local pool: %s", txHash)
+	}
+	// The sequence should not be incremented if the transaction was evicted
```

Comment:
```suggestion
	// The sequence should not be rolled back to the sequence of the transaction that was evicted to be
	// ready for resubmission. All transactions with a later nonce will be kicked by the nodes tx pool.
```
nit: some more context here would be helpful